apps/test/unit/codebridge/FileBrowser/prompts/openImportFromBackpackPromptTest.ts

```diff
@@ -0,0 +1,199 @@
+    projectFiles: MultiFileSource['files'];
+
+  beforeEach(() => {
+    mockBackpackApi = getBackpackAPIMock(); // getFileList returns empty list.
+    dialogControl = {
+      showDialog: jest.fn(),
+    };
+    newFileFunction = jest.fn();
+    saveFileFunction = jest.fn();
+    projectFiles = {
+      '1': {
+        id: '1',
+        name: 'project_file1.py',
+        language: 'py',
+        contents: 'This is project_file1.py',
+        folderId: '0',
+      },
+    };
+  });
+
+  const runImportFromBackpackPrompt = async (extraArgs = {}) => {
+    await openImportFromBackpackPrompt({
+      dialogControl,
+      backpackApi: mockBackpackApi,
+      newFile: newFileFunction,
+      saveFile: saveFileFunction,
+      projectFiles,
+      ...extraArgs,
+    });
+  };
+
+  it('shows an alert when there are no files in the backpack', async () => {
+    dialogControl.showDialog.mockResolvedValue({
+      type: 'confirm',
+    });
+
+    await runImportFromBackpackPrompt();
+
+    expect(mockBackpackApi.getFileList).toHaveBeenCalled();
+    expect(dialogControl.showDialog).toHaveBeenCalledWith(
+      expect.objectContaining({
+        type: 'GenericAlert',
+        title: expect.any(String),
+        message: expect.any(String),
+      })
+    );
+  });
+
+  it('imports a backpack file when user confirms', async () => {
+    mockBackpackApi = getBackpackAPIMock(['backpack_file.py']); // getFileList returns ['backpack_file.py'].
+    (extractUserInput as jest.Mock).mockReturnValue('backpack_file.py');
+    // Mock the dialog responses
+    dialogControl.showDialog
+      .mockResolvedValueOnce({}) // Pending dialog.
+      .mockResolvedValueOnce({type: 'confirm'}); // User confirms import.
+
+    await runImportFromBackpackPrompt();
+
+    expect(mockBackpackApi.getFileList).toHaveBeenCalled();
+    expect(mockBackpackApi.fetchFile).toHaveBeenCalledWith(
+      'backpack_file.py',
+      expect.any(Function),
+      expect.any(Function)
+    );
+    expect(newFileFunction).toHaveBeenCalledWith({
+      contents: 'Mock contents of backpack file backpack_file.py',
+      fileName: 'backpack_file.py',
+    });
+  });
+
+  it('renames imported backpack file if duplicate exists and user selects to rename', async () => {
+    mockBackpackApi = getBackpackAPIMock(['test_file.py']);
+    projectFiles = {
+      '1': {
+        id: '1',
+        name: 'test_file.py',
+        language: 'py',
+        contents: 'This is project test_file.py',
+        folderId: '0',
+      },
+    };
+    dialogControl.showDialog
+      .mockResolvedValueOnce({}) // Pending dialog.
+      .mockResolvedValueOnce({type: 'confirm'}) // User confirms import.
+      .mockResolvedValueOnce({type: 'neutral'}); // User chooses to rename.
+
+    (extractUserInput as jest.Mock).mockReturnValue('test_file.py');
+
+    await runImportFromBackpackPrompt();
+
+    // Ensure async calls complete before assertions
+    await new Promise(resolve => setTimeout(resolve, 100));
```

Comment:
I think you can use [waitFor](https://testing-library.com/docs/dom-testing-library/api-async/#waitfor) instead of putting a timeout here.

Reply:
Cool! Updated.